pkg/cloudprovider/providers/gce/gce_loadbalancer_naming.go

```diff
@@ -33,15 +35,16 @@ func makeInstanceGroupName(clusterID string) string {
 
 func makeBackendServiceName(loadBalancerName, clusterID string, shared bool, scheme lbScheme, protocol v1.Protocol, svcAffinity v1.ServiceAffinity) string {
 	if shared {
-		affinity := ""
-		switch svcAffinity {
-		case v1.ServiceAffinityClientIP:
-			affinity = "clientip"
-		default:
-			affinity = "noaffinity"
-		}
-
-		return fmt.Sprintf("k8s-%s-%s-%s-%s", clusterID, strings.ToLower(string(scheme)), strings.ToLower(string(protocol)), affinity)
+		hash := sha1.New()
+
+		// For every non-nil option, hash its value. Currently, only service affinity is relevant.
+		hash.Write([]byte(string(svcAffinity)))
+
+		hashed := hex.EncodeToString(hash.Sum(nil))
+		hashed = hashed[:16]
+
+		// 3 + 1 + 16 + 1 + 8 + 1 + 3 + 16
```

Comment:
maybe document this a bit more than just the numbers?

http://lostpedia.wikia.com/wiki/The_Numbers